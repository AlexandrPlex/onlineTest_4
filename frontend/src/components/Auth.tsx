import * as hash from 'object-hash';
import * as React from 'react';

export interface IAuthProps {
  authState: boolean;
}

export interface IAuthDispatch {
    login: (email: string, password: string) => Promise<any>;
}

export class Auth extends React.Component<IAuthProps & IAuthDispatch> {
    public state = {
        emailValue: '',
        passwordValue: '',
    };
    public onChangeInput = (e: React.FormEvent<HTMLInputElement>) => {
        const state: any = {...this.state};
        state[e.currentTarget.name] = e.currentTarget.name === 'passwordValue' ? hash(e.currentTarget.value) : e.currentTarget.value;
        this.setState(state);
    }
    public onHandleSubmit = () => {
        alert(this.props.login(this.state.emailValue, this.state.passwordValue));
    }
    public render() {
        return (
            <React.Fragment>
                <form onSubmit={ (e: React.FormEvent<HTMLFormElement>) => { e.preventDefault(); } }>
                    <input onChange={this.onChangeInput} type='text' name='loginValue' placeholder='user name' />
                    <input onChange={this.onChangeInput} type='password' name='passwordValue' placeholder='password' />
                    <input onClick={this.onHandleSubmit} type='submit' name='submit'/>
                </form>
            </React.Fragment>
        );
    }
}
