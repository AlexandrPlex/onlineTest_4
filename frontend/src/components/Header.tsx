import * as React from 'react';
import { NavLink } from 'react-router-dom';

interface IHeaderProps {
  dropdownLists: string[];
}

export class Header extends React.Component<IHeaderProps> {
  public render() {
    return (
      <nav>
        <div className='nav-wrapper'>
          <div className='container'>
            <span className='brand-logo'><NavLink exact={true} activeClassName='active-link' to='/'>Logo</NavLink></span>
            <ul id='nav-mobile' className='right hide-on-med-and-down'>
              <li key='react'><NavLink activeClassName='active-link' to='/react'>React</NavLink></li>
              <li key='materialize'>
                <a className='dropdown-button' href='#' data-target='header-dropdown'>Dropdown</a>
              </li>
              <li key='404'><NavLink activeClassName='active-link' to='/404'>NotFound</NavLink></li>
            </ul>
          </div>
        </div>
      </nav>
    );
  }
}
