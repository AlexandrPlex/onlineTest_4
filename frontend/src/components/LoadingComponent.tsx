import * as React from 'react';

export class LoadingComponent extends React.PureComponent {
  public render() {
    return (
      <React.Fragment>
          <h1>LOADING</h1>
      </React.Fragment>
    );
  }
}
