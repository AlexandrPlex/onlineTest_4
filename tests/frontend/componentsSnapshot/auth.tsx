import {Auth, IAuthDispatch} from 'components/Auth';
import React from 'react';
import renderer from 'react-test-renderer';

describe('[SnapShot] Auth test', () => {
    it(' - Auth', ( ) => {
        const state: IAuthDispatch = {
            login: () => 0,
        };
        const tree = renderer.create( <Auth { ...state }/>);
        expect(tree).toMatchSnapshot();
    });
});
