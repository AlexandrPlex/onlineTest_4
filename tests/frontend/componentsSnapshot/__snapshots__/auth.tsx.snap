// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SnapShot] Auth test  - Auth 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="emailValue"
    onChange={[Function]}
    placeholder="user name"
    type="text"
  />
  <input
    name="passwordValue"
    onChange={[Function]}
    placeholder="password"
    type="password"
  />
  <input
    name="submit"
    onClick={[Function]}
    type="submit"
  />
</form>
`;
